core/src/services/resizeObserver.ts

```diff
@@ -1,45 +1,79 @@
 import type {ReadableSignal} from '@amadeus-it-group/tansu';
 import {derived} from '@amadeus-it-group/tansu';
-import {createBrowserStoreDirective} from '../utils/directive';
+import {createBrowserStoreArrayDirective} from '../utils/directive';
 import {noop} from '../utils/func';
 import type {Directive, SSRHTMLElement} from '../types';
 
 /**
- * Create a resize observer object
- * @returns An object containing the store with the dimentions of observed element (ResizeObserverEntry), the directive to be applied to the html element to be observed
+ * Create a resize observer that can be applied to multiple elements
+ * @param maxElements - Maximum number of elements allowed (undefined = unlimited)
+ * @returns An object containing the store with the dimensions map of observed elements (ResizeObserverEntry), the directive to be applied to the html element to be observed
  */
-export const createResizeObserver = (): {
-	dimensions$: ReadableSignal<ResizeObserverEntry | undefined>;
+export const createResizeObserverArray = (
+	maxElements?: number,
+): {
+	dimensionsMap$: ReadableSignal<Map<HTMLElement, ResizeObserverEntry>>;
 	directive: Directive<void, SSRHTMLElement>;
 } => {
-	const {element$, directive} = createBrowserStoreDirective();
+	const {elements$, directive: arrayDirective} = createBrowserStoreArrayDirective();
+	const registeredElements = new Set<SSRHTMLElement>();
+
+	const directive: Directive<void, SSRHTMLElement> = (element) => {
+		if (maxElements !== undefined && registeredElements.size >= maxElements && !registeredElements.has(element)) {
```

Comment:
I think you should not check whether the element is already present. Calling multiple times the directive with the same element can count as multiple usages.